src/arrayMethodSort.js

```diff
@@ -4,8 +4,36 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    if (compareFunction === undefined) {
+      for (let i = 0; i < this.length; i++) {
+        for (let k = 0; k < this.length; k++) {
+          if ('' + this[k] > '' + this[k + 1]) {
+            const temp = this[k];
+
+            this[k] = this[k + 1];
+            this[k + 1] = temp;
+          }
+        }
+      }
+
+      return this;
+    }
+
+    if (compareFunction) {
+      for (let i = 0; i < this.length; i++) {
+        for (let k = 0; k < this.length; k++) {
+          if (compareFunction(this[k], this[k + 1]) >= 1) {
```

Comment:
The comparison function should return a negative value if the first argument is less than the second, zero if they're equal, and a positive value if the first is greater. The condition should check for 'compareFunction(this[k], this[k + 1]) > 0' instead of '>= 1'.